Cwx.m

```diff
@@ -0,0 +1,360 @@
+    for (int i=0 ; i < MAX_NUMBER_OF_MACROS ; i++) {
+        [_macros addObject: @""];
+    }
+    [self initcwxTokens];
+    
+    _cwxRunQueue = dispatch_queue_create("net.k6tu.cwx", DISPATCH_QUEUE_SERIAL);
+    
+    return self;
+}
+
+
+
+- (void) initcwxTokens {
+  self.cwxTokens = [[NSDictionary alloc] initWithObjectsAndKeys:
+                          [NSNumber numberWithInteger:breakinDelay], @"break_in_delay",
+                          [NSNumber numberWithInteger:eraseSent], @"erase",
+                          [NSNumber numberWithInteger:macro], @"macro",
+                          [NSNumber numberWithInteger:sent], @"sent",
+                          [NSNumber numberWithInteger:wpm], @"wpm",
+                          nil];
+}
+
+// Private macro to improve readibility of setters
+
+#define commandUpdateNotify(cmd, key, ivar, value) \
+/* Let observers know the change on the main queue */ \
+[self willChangeValueForKey:(key)]; \
+(ivar) = (value); \
+[self didChangeValueForKey:(key)]; \
+\
+__weak Cwx *safeSelf = self; \
+dispatch_async(self.cwxRunQueue, ^(void) { \
+/* Send the command to the radio on our private queue */ \
+[safeSelf.radio commandToRadio:(cmd)]; \
+});
+
+
+- (void) setDelay:(int)delay {
+    if (delay < 0) delay = 0;
+    if (delay > MAX_CWX_DELAY_MS) delay = MAX_CWX_DELAY_MS;
+//    if (_delay == delay) return;
+    
+    NSString *cmd = [NSString stringWithFormat:@"cwx delay %i", delay];
+    
+    commandUpdateNotify(cmd, @"delay", _delay, delay);
+}
+
+
+- (void) setSpeed:(int)speed {
+    if (speed < MIN_CWX_SPEED) speed = MIN_CWX_SPEED;
+    if (speed > MAX_CWX_SPEED) speed = MAX_CWX_SPEED;
+//    if (_speed == speed) return;
+    
+    NSString *cmd = [NSString stringWithFormat:@"cwx wpm %i", speed];
+    
+    commandUpdateNotify(cmd, @"speed", _speed, speed);
+}
+
+
+- (bool) setMacro:(int)index macro:(NSString *) msg {
+    if (index < 0 || index > MAX_NUMBER_OF_MACROS - 1) return false;
+    
+    NSString *cmd = [NSString stringWithFormat:@"cwx macro save %i \"%@\"" , index + 1, msg];
+    
+    NSMutableArray * macrosRef = _macros;
+    macrosRef[index] = msg;
+    commandUpdateNotify(cmd, @"macros", _macros, macrosRef);
+    return true;
+}
+
+
+- (bool) getMacro:(int)index macro:(NSString **)string {
+    *string = @"";
+    if (index < 0 || index > MAX_NUMBER_OF_MACROS - 1) return false;
+    *string = (NSString *)_macros[index];
+    return true;
+}
+
+
+- (int) sendMacro:(int) index {
+    if (index < 0 || index > MAX_NUMBER_OF_MACROS - 1) return 0;
+    NSString *cmd = [NSString stringWithFormat:@"cwx macro send %i", index + 1];
+    return [_radio commandToRadio:cmd notify:self];
+}
+
+
+- (void) clearBuffer {
+    [_radio commandToRadio:@"cwx clear"];
+}
+
+
+- (void) erase:(int)numberOfChars {
+    NSString *cmd = [NSString stringWithFormat:@"cwx erase %i", numberOfChars];
+    [_radio commandToRadio:cmd ];
+}
+
+
+- (double) getTxFrequency {
```

Comment:
Hmmm....  model decoupling suggest that this doesn't belong here.  I think this breaks information partitioning between radio, slices and CWX.  Suggest that if this belongs anywhere, it should go on Radio and not on CWX.